File: data/scripts/process_data.py
import json
import re

def process():
    # 1. 加载分类信息
    category_map = {}
    with open('data/category-info.txt', 'r', encoding='utf-8') as f:
        lines = f.readlines()
        if lines:
            for line in lines[1:]:
                parts = line.strip().split('\t')
                if len(parts) >= 2:
                    category_id = parts[0]
                    category_name = parts[1]
                    category_map[category_id] = category_name

    # 2. 加载数据文件
    with open('data/理论资源导航及子孙栏目数据.txt', 'r', encoding='utf-8') as f:
        data = json.load(f)

    # 3. 处理并输出 Markdown 表格
    results = []
    results.append("| 标题 | 地址 URL | 分类名称 |")
    results.append("| :--- | :--- | :--- |")
    
    link_pattern = re.compile(r"<a href=['\"](.*?)['\"].*?>(.*?)</a>")
    
    for item in data:
        source = item.get('_source', {})
        show_title_html = source.get('showTitle', '')
        category_id = source.get('categoryId', '')
        
        match = link_pattern.search(show_title_html)
        if match:
            url = match.group(1)
            title = match.group(2)
            category_name = category_map.get(category_id, '未知分类')
            # 过滤掉“头图及栏目图”之类可能干扰的条目（如果有残留）
            if category_name != "头图及栏目图":
                results.append(f"| {title} | {url} | {category_name} |")

    with open('output_navigation.md', 'w', encoding='utf-8') as f:
        f.write('\n'.join(results))

if __name__ == "__main__":
    process()

File: web/scripts/process_data.js
const fs = require('fs');
const path = require('path');

const inputFile = path.join(__dirname, '../data/output_list.txt');
const outputFile = path.join(__dirname, '../src/data/sites.json');

try {
    const data = fs.readFileSync(inputFile, 'utf-8');
    const lines = data.split('\n');
    const sites = [];

    // Skip header and separator (first 2 lines)
    // Also filtering out empty lines
    let idCounter = 1;

    for (let i = 2; i < lines.length; i++) {
        const line = lines[i].trim();
        if (!line) continue;

        // Split by tab
        const parts = line.split('\t');

        // Ensure we have enough parts. 
        // Based on view_file data/output_list.txt, there are 4 columns.
        // Some lines might have missing descriptions or extra tabs.
        // The format seems to be: Title, URL, Category, Description

        if (parts.length >= 3) {
            const title = parts[0].trim();
            const url = parts[1].trim();
            const category = parts[2].trim();
            const description = parts[3] ? parts[3].trim() : '';

            // Basic validation
            if (title && url) {
                sites.push({
                    id: idCounter++,
                    name: title,
                    url: url,
                    category: category,
                    desc: description,
                    tags: [], // Placeholder for tags
                    icon: title.charAt(0) // Default icon: first char
                });
            }
        }
    }

    // Extract unique categories for verification (optional logging)
    const categories = [...new Set(sites.map(s => s.category))];
    console.log(`Processed ${sites.length} sites.`);
    console.log(`Found ${categories.length} categories:`, categories);

    // Group sites? No, the frontend will filter/group.
    // Just write the flat list.

    fs.writeFileSync(outputFile, JSON.stringify(sites, null, 2), 'utf-8');
    console.log(`Successfully wrote to ${outputFile}`);

} catch (err) {
    console.error('Error processing data:', err);
}
